Content.Server/Foldable/FoldableSystem.cs

```diff
@@ -0,0 +1,199 @@
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+
+namespace Content.Server.Foldable
+{
+    [UsedImplicitly]
+    public sealed class FoldableSystem : EntitySystem
+    {
+        [Dependency] private readonly InteractionSystem _interactionSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<FoldableComponent, InteractHandEvent>(OnInteract);
+            SubscribeLocalEvent<FoldableComponent, PickedUpAttemptEvent>(OnPickedUpAttempt);
+            SubscribeLocalEvent<FoldableComponent, UseInHandEvent>(OnUse);
+
+            SubscribeLocalEvent<FoldableComponent, GetInteractionVerbsEvent>(AddFoldVerb);
+        }
+
+        private bool TryToggleFold(FoldableComponent comp)
+        {
+            return TrySetFolded(comp, !comp.IsFolded);
+        }
+
+        /// <summary>
+        /// Try to fold/unfold
+        /// </summary>
+        /// <param name="comp"></param>
+        /// <param name="state">Folded state we want</param>
+        /// <returns>True if successful</returns>
+        private bool TrySetFolded(FoldableComponent comp, bool state)
+        {
+            if (state == comp.IsFolded)
+                return false;
+
+            if (comp.Owner.IsInContainer())
+                return false;
+
+            // First we check if the foldable object has a strap component
+            if (comp.Owner.TryGetComponent(out StrapComponent? strap))
+            {
+                // If an entity is buckled to the object we can't pick it up or fold it
+                if (strap.BuckledEntities.Any())
+                    return false;
+            }
+
+            SetFolded(comp, state);
+            return true;
+        }
+
+        /// <summary>
+        /// Set the folded state of the given <see cref="FoldableComponent"/>
+        /// </summary>
+        /// <param name="component"></param>
+        /// <param name="folded">If true, the component will become folded, else unfolded</param>
+        private void SetFolded(FoldableComponent component, bool folded)
+        {
+            component.IsFolded = folded;
+            component.CanBeFolded = !component.Owner.IsInContainer();
+
+            // You can't buckle an entity to a folded object
+            if (component.Owner.TryGetComponent(out StrapComponent? strap))
+                strap.Enabled = !component.IsFolded;
+
+            // A folded object is collision-less
+            if (component.Owner.TryGetComponent(out PhysicsComponent? physicsComponent))
+                physicsComponent.CanCollide = !component.IsFolded;
+
+            // Update visuals only if the value has changed
+            if (component.Owner.TryGetComponent(out AppearanceComponent? appearance))
+                appearance.SetData("FoldedState", folded);
+        }
+
+        /// <summary>
+        /// Tries to deploy the foldable in front of the user
+        /// </summary>
```

Comment:
Why doesn't this just deploy on its position?

Reply:
Because otherwise it will just push the user around?

Reply:
I think you're better off just toggling the mask whenever something is buckled or not.